src/SPIN/fix_nve_spin.cpp

```diff
@@ -172,15 +175,18 @@ void FixNVESpin::init()
 
   // loop 1: obtain # of Pairs, and # of Pair/Spin styles
 
-  if (force->pair_match("spin",0,0)) {        // only one Pair/Spin style
-    pair = force->pair_match("spin",0,0);
-    npairs = pair->instance_total;
```

Comment:
@julient31 I noticed this while improving the inexact matches. `Pair::instance_total` is **not** the total number of pair styles but the total number of pair styles **ever created**. This is to give a unique number to identify pair styles that are otherwise the same, e.g. when using the same pair style as a hybrid sub-style. This is used for example by the neighbor list code to by able to distinguish individual neighbor list requests. So if you replace an initial pair style (say lj/cut with morse) its number will become 2 even though there is only one pair style instance active.